examples/lines_bars_and_markers/scatter_with_legend.py

```diff
@@ -1,3 +1,11 @@
+'''
+Demo of a scatter plot (scatter) with a legend. 
+
+Also demonstrates how transparency of the markers 
+can be adjusted by giving ``alpha`` a value between 
+0 and 1.
```

Comment:
Can we please showcase this paragraph as an *ideal* example of an example docstring? Clear and concise, and highlighted exactly what this example demonstrates!